### PR TITLE
Update the target sdk version and library versions

### PR DESCRIPTION
**Pull request by thagikura**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -19,23 +19,23 @@
...
+        targetSdkVersion = 29
+        compileSdkVersion = 29
 
-        androidGradlePluginVersion = "3.2.1"
+        androidGradlePluginVersion = "3.5.0"
         androidMavenGradlePluginVersion = "1.5"
-        androidxAnnotationVersion = "1.0.0"
-        androidxAppCompatVersion = "1.0.2"
-        androidxCoreVersion = "1.0.1"
+        androidxAnnotationVersion = "1.0.2"
```
Comment (MGaetan89) [inline build.gradle]: The latest version is `1.1.0`.

File: build.gradle [hunk 2]
```diff
@@ -19,23 +19,23 @@
...
+        androidxCoreVersion = "1.1.0"
         androidxLegacySupportV4Version = "1.0.0"
-        androidxPreferenceVersion = "1.0.0"
+        androidxPreferenceVersion = "1.1.0"
         androidxPreferenceV14Version = "1.0.0"
         androidxRecyclerViewVersion = "1.0.0"
-        androidxEspressoVersion = "3.1.0"
-        androidxTestVersion = "1.1.0-alpha4"
+        androidxEspressoVersion = "3.2.0"
+        androidxTestVersion = "1.1.1"
```
Comment (MGaetan89) [inline build.gradle]: The latest version is `1.2.0`.

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Mon Sep 24 09:09:16 CEST 2018
+#Mon Sep 09 14:42:19 PDT 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
```
Comment (MGaetan89) [inline gradle/wrapper/gradle-wrapper.properties]: The latest version is `5.6.2`.

File: build.gradle [hunk 4]
```diff
@@ -19,23 +19,23 @@
...
-        androidGradlePluginVersion = "3.2.1"
+        androidGradlePluginVersion = "3.5.0"
         androidMavenGradlePluginVersion = "1.5"
-        androidxAnnotationVersion = "1.0.0"
-        androidxAppCompatVersion = "1.0.2"
-        androidxCoreVersion = "1.0.1"
+        androidxAnnotationVersion = "1.0.2"
+        androidxAppCompatVersion = "1.1.0"
+        androidxCoreVersion = "1.1.0"
         androidxLegacySupportV4Version = "1.0.0"
```
Comment (MGaetan89) [inline build.gradle]: I think this one is only declared and included [here](https://github.com/google/flexbox-layout/blob/update_to_29/demo-playground/build.gradle#L43), but never used.

File: build.gradle [hunk 5]
```diff
@@ -19,23 +19,23 @@
...
-        androidxPreferenceVersion = "1.0.0"
+        androidxPreferenceVersion = "1.1.0"
         androidxPreferenceV14Version = "1.0.0"
```
Comment (MGaetan89) [inline build.gradle]: I think this one is only declared and included [here](https://github.com/google/flexbox-layout/blob/update_to_29/demo-playground/build.gradle#L46), but never used.

File: build.gradle (hunk 4 above)
Comment (thagikura) [reply build.gradle]: Done

File: build.gradle (hunk 1 above)
Comment (thagikura) [reply build.gradle]: Done

File: build.gradle (hunk 2 above)
Comment (thagikura) [reply build.gradle]: Done

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (thagikura) [reply gradle/wrapper/gradle-wrapper.properties]: Done

File: build.gradle (hunk 5 above)
Comment (thagikura) [reply build.gradle]: Done

**Comment by thagikura** [comment]
Addressed. Thanks!

**Pull request merged**